packages/flutter_tools/lib/src/reporting/events.dart

```diff
@@ -248,67 +248,3 @@ class CodeSizeEvent extends UsageEvent {
 class ErrorHandlingEvent extends UsageEvent {
   ErrorHandlingEvent(String parameter) : super('error-handling', parameter, flutterUsage: globals.flutterUsage);
 }
-
-/// Emit various null safety analytic events.
-///
-/// 1. The current null safety runtime mode.
-/// 2. The number of packages that are migrated, along with the total number of packages
-/// 3. The main packages language version.
-class NullSafetyAnalysisEvent implements UsageEvent {
-  NullSafetyAnalysisEvent(
-    this.packageConfig,
-    this.nullSafetyMode,
-    this.currentPackage,
-    this.flutterUsage,
-  );
-
-  /// The category for analytics events related to null safety.
-  static const String kNullSafetyCategory = 'null-safety';
-
-  final PackageConfig packageConfig;
-  final NullSafetyMode nullSafetyMode;
-  final String currentPackage;
-  @override
-  final Usage flutterUsage;
-
-  @override
-  void send() {
-    if (packageConfig.packages.isEmpty) {
-      return;
-    }
-    int migrated = 0;
-    LanguageVersion? languageVersion;
-    for (final Package package in packageConfig.packages) {
-      final LanguageVersion? packageLanguageVersion = package.languageVersion;
-      if (package.name == currentPackage) {
-        languageVersion = packageLanguageVersion;
-      }
-      if (packageLanguageVersion != null &&
-          packageLanguageVersion.major >= nullSafeVersion.major &&
-          packageLanguageVersion.minor >= nullSafeVersion.minor) {
-        migrated += 1;
-      }
-    }
-    flutterUsage.sendEvent(kNullSafetyCategory, 'runtime-mode', label: nullSafetyMode.toString());
-    flutterUsage.sendEvent(kNullSafetyCategory, 'stats', parameters: CustomDimensions(
-      nullSafeMigratedLibraries: migrated,
```

Comment:
Can these fields be removed from the `CustomDimensions` class?

Reply:
ooh, good catch

Reply:
WDYT about this diff https://github.com/flutter/flutter/pull/120936/commits/3519e1f387ce6c7e46412f1fbd65c1cc00015223